File: configuration/src/test/java/com/synopsys/integration/configuration/property/types/path/PathValueTest.java
package com.synopsys.integration.configuration.property.types.path;

import java.nio.file.Path;
import java.nio.file.Paths;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class PathValueTest {
    private static class TestPathResolver implements PathResolver {
        @Override
        public Path resolvePath(String filePath) {
            return Paths.get(filePath);
        }
    }

    @Test
    public void resolvePathTest() {
        Assertions.assertEquals(Paths.get("/test"), new PathValue("/test").resolvePath(new TestPathResolver()));
    }

    @Test
    public void toStringTest() {
        Assertions.assertEquals("/test/toString", new PathValue("/test/toString").toString());
    }
}

File: configuration/src/test/java/com/synopsys/integration/configuration/property/types/path/PathValueTest.kt
package com.synopsys.integration.configuration.property.types.path

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.nio.file.Path
import java.nio.file.Paths

internal class PathValueTest {
    private class TestPathResolver : PathResolver {
        override fun resolvePath(filePath: String): Path {
            return Paths.get(filePath)
        }
    }

    @Test
    fun resolvePathTest() {
        Assertions.assertEquals(Paths.get("/test"), PathValue("/test").resolvePath(TestPathResolver()))
    }

    @Test
    fun toStringTest() {
        Assertions.assertEquals("/test/toString", PathValue("/test/toString").toString())
    }
}